src/services/recipient.js

```diff
@@ -349,25 +349,33 @@ export async function recipientsByName(query, scopes, sortBy, direction, offset,
  * So instead, we depuplicating once after the objectives have been reduced, and accounting for
  * the differing formats then
  */
-function reduceTopicsOfDifferingType(topics) {
-  const newTopics = uniq(topics.map((topic) => {
-    if (typeof topic === 'string') {
-      return topic;
-    }
+export function reduceTopicsOfDifferingType(topics) {
+  const newTopics = uniq((topics || null)
+    .filter((topic) => topic !== null && topic !== undefined).map((topic) => {
+      if (typeof topic === 'string') {
+        return topic;
+      }
 
-    if (topic.name) {
-      return topic.name;
-    }
+      if (typeof topic === 'object' && topic.name) {
+        return topic.name.trim();
+      }
 
-    return topic;
-  }));
+      return topic;
+    }));
 
-  newTopics.sort();
+  newTopics.sort((a, b) => {
+    if (typeof a === 'string' && typeof b === 'string') {
+      return a.localeCompare(b);
+    }
+    if (typeof a === 'string') return -1;
+    if (typeof b === 'string') return 1;
+    return 0;
+  });
```

Comment:
This wont reduce it as expected, the unique or similar functionality needs to be last or twice. topcis = ['apples', { name: "apples"}] => ["apples", "apples"]

Reply:
We are good here as is for the case `topics = ['apples', { name: "apples"}] => ["apples", "apples"]`. The second 'apples' topic is already handled by this logic:

```
      if (typeof topic === 'object' && topic.name) {
        return topic.name.trim();
      }
```
and deduplicated via `uniq`

Additionally, the existing test case on line 1024,` it('should handle mixed strings and objects with "name"', () => {` already tests and confirms it.

Reply:
I see the issue with how I was reading it now. the map is within the uniq.

Reply:
The values from the string path do still need to be trimmed

Reply:
> Its interesting that you are using multiple methods to unique the values, lodash.unique and [...new Set()].

Good catch! I prefer using native JavaScript functions like Set to avoid relying on external libraries, but since the code works fine as-is, I don’t see a big reason to refactor it right now. If we end up revisiting this file later, I’d be up for cleaning it up to use native methods.

Reply:
> The values from the string path do still need to be trimmed

`trim()` was added.